### PR TITLE
Remove DatePicker and PickerWindows

### PR DESCRIPTION
**Pull request by NickGerleman**
This removes some lean-core components that were Windows specific and previously warned about deprecation. `PickerWindows`  source is not fully removed since the inbuilt Picker relies on it, and has not yet been removed. We instead just no longer expose the Windows specific API to instead move people towards the community module.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7685)

### CODE REVIEW

File: vnext/src/index.windows.js [hunk 1]
```diff
@@ -543,8 +540,12 @@ module.exports = {
       .supportKeyboard;
   },
   get DayOfWeek(): any {
-    return require('./Libraries/Components/DatePicker/DatePickerProps')
-      .DayOfWeek;
+    invariant(
```
Comment (NickGerleman) [inline vnext/src/index.windows.js:543]: Need to see if flow recognizes this is no-return

**Comment by asklar** [comment]
@NickGerleman we need to make sure certain internal partners are ok with this

**Comment by NickGerleman** [comment]
@asklar would be eager to know how that goes. 0.64 yellowboxes on usage, since we want to move away from having two DateTimePickers. We have not been putting new APIs fixes into the built-in one, so we want to point people to the non-deprecated one.

A strategy some internal teams have used before is to do a snapshot-in-time source copy of the community module.

**Comment by NickGerleman** [comment]
At the same time, Removing `acceptsKeyboardFocus` a release after yellowboxing was a mess. One of the ideas we had was to space out by another release. So this one might be better in 0.66

**Comment by NickGerleman** [comment]
Going to keep this around as a draft for when we do want to use it.

**Comment by NickGerleman** [comment]
Going to have some offline discussion about deprecation policy in general. This one should end up being less painful than acceptsKeyboardFocus since alternatives have been around for a few releases.

**Comment by NickGerleman** [comment]
Exiting draft. Merging now would mean this change comes in 0.66.

**Comment by NickGerleman** [comment]
cc @acoates-ms @asklar @rectified95 

Picker itself is removed very shortly after our current nightly build, so this avoids doing as much in the integration PR (since PickerWindows relies on Picker).

**Review by acoates-ms — APPROVED** [review]

**Pull request merged**